### ISSUE TITLE
Example doesn't sync audio + video with ffmpeg.exe

### ISSUE THREAD

**Issue by jabrena** [muaz-khan/RecordRTC#48]
Hi,

I am testing the project in a Windows environment with latest ffmpeg (ffmpeg-20151008-git-5911eeb-win64-static) but in local, I noticed that audio has some delay when ffmpeg mix audio & video.

Any idea?

**Comment by Rsgm** [comment]
I am experiencing the same issue with my project. I use [MediaStreamRecorder](https://github.com/streamproc/MediaStreamRecorder), so I think it has to do with the browser. I have noticed it is was a bit worse on chrome than firefox.

I record a video from the webcam while uploading blobs at a set interval and compile them on the server. I have noticed that it is usually synced up, especially with smaller videos. When it does end up desyncing, it is somewhat random, but usually on longer videos. It seems to happen at a certain random point in the video as well. That is, it may start to desync at the 2 minute mark or 3 minute mark of a 5 minute video or possibly right a way.

Anyway, this is what I have experienced, I hope it helps even though I use MediaStreamRecorder.

**Comment by jabrena** [comment]
Totally agree with you.
In a preliminar iteration, I used this example:
https://github.com/muaz-khan/RecordRTC/tree/master/RecordRTC-to-Nodejs

but finally, I used this combination:
https://github.com/collab-project/videojs-record

to upload a .webm file to server using express.
https://github.com/jabrena/CloudFoundryLab/tree/master/Node_Upload_file

I think that the unique web browser which it generate a real video in client side is Firefox.

Juan Antonio

**Comment by jostheeuwen** [comment]
I had this issue (or something similar) in Firefox. The audio and video would get out of sync if I used ffmpeg to transcode the webm from FF to an mp4. When manually running the transcoding I would get a lot of warnings about the audio timestamps:

`[libvo_aacenc @ 05583c40] Queue input is backward in time
[mp4 @ 05593860] Non-monotonous DTS in output stream 0:1; previous: 1201058, current: 1200512; changing to 1201059. This may result in incorrect timestamps in the output file.`

I eventually fixed it by recalculating the timestamps for the audio by using an ffmpeg filter:

`ffmpeg -i input.webm -filter:a "asetpts=N/SR/TB" output.mp4`

More background info is on the ffmpeg website:
https://ffmpeg.org/ffmpeg-filters.html#setpts_002c-asetpts

Hope this helps.
Jos
